Bio/Affy/CelFile.py

```diff
@@ -332,64 +332,65 @@ def _read_v3(handle):
+                    _, filename = line.replace("\x14", "")[:index].split()
                     record.DatHeader["filename"] = filename
-                    index += 1
-                    field = line[index : index + 9]
-                    assert field[:4] == "CLS="
-                    assert field[8] == " "
-                    record.DatHeader["CLS"] = int(field[4:8])
-                    index += 9
-                    field = line[index : index + 9]
-                    assert field[:4] == "RWS="
-                    assert field[8] == " "
-                    record.DatHeader["RWS"] = int(field[4:8])
-                    index += 9
-                    field = line[index : index + 7]
-                    assert field[:4] == "XIN="
-                    assert field[6] == " "
-                    record.DatHeader["XIN"] = int(field[4:6])
-                    index += 7
-                    field = line[index : index + 7]
-                    assert field[:4] == "YIN="
-                    assert field[6] == " "
-                    record.DatHeader["YIN"] = int(field[4:6])
-                    index += 7
-                    field = line[index : index + 6]
-                    assert field[:3] == "VE="
-                    assert field[5] == " "
-                    record.DatHeader["VE"] = int(field[3:5])
-                    index += 6
-                    field = line[index : index + 7]
-                    assert field[6] == " "
-                    temperature = field[:6].strip()
-                    if temperature:
-                        record.DatHeader["temperature"] = int(temperature)
-                    else:
-                        record.DatHeader["temperature"] = None
-                    index += 7
-                    field = line[index : index + 4]
-                    assert field.endswith(" ")
-                    record.DatHeader["laser-power"] = float(field)
-                    index += 4
-                    field = line[index : index + 18]
-                    assert field[8] == " "
-                    record.DatHeader["scan-date"] = field[:8]
-                    assert field[17] == " "
-                    record.DatHeader["scan-time"] = field[9:17]
-                    index += 18
-                    field = line[index:]
-                    subfields = field.split(" \x14 ")
-                    assert len(subfields) == 12
-                    subfield = subfields[0]
-                    try:
-                        scanner_id, scanner_type = subfield.split()
-                    except ValueError:
-                        scanner_id = subfield.strip()
-                    record.DatHeader["scanner-id"] = scanner_id
-                    record.DatHeader["scanner-type"] = scanner_type
-                    record.DatHeader["array-type"] = subfields[2]
-                    record.DatHeader["image-orientation"] = int(subfields[11])
+                    if index > 0:
```

Comment:
I understand ``index = -1`` means no ``:`` was found.

Can we have ``index = 0`` meaning ``line.startswith(":")``? Perhaps that should raise a ``ValueError``?

Reply:
Exactly, the file I am looking at starts like this:
```
[CEL]
Version=3

[HEADER]
Cols=744
Rows=744
TotalX=744
TotalY=744
OffsetX=0
OffsetY=0
GridCornerUL=0 0
GridCornerUR=743 0
GridCornerLR=743 743
GridCornerLL=0 743
Axis-invertX=0
AxisInvertY=0
swapXY=0
DatHeader=   HG-U219.1sq          
Algorithm=Percentile
AlgorithmParameters=Percentile:75;CellMargin:2;OutlierHigh:1.500000;OutlierLow:1.004000;NumPixelsToUse:0;ImageCalibration:FALSE;FeatureExtraction:TRUE;IgnoreShiftRowOutliers:FALSE;FixedCellSize:TRUE;UseSubgrids:FALSE;RandomizePixels:FALSE;ErrorBasis:StdvMean;PercentileSpread:15.000000;StdMult:1.000000;ExtendPoolWidth:2;ExtendPoolHeight:2;OutlierRatioLowPercentile:55;OutlierRatioHighPercentile:75;HalfCellRowsDivisor:5;HalfCellRowsRemainder:4;CellIntensityCalculationType:Percentile;HighCutoff:3500;LowCutoff:4096;PairCutoff:2.500000;featureRows:109;featureColumns:127;featureWidth:8.000000;featureHeight:8.000000;FullFeatureWidth:8;FullFeatureHeight:8

[INTENSITY]
NumberCells=553536
CellHeader=X	Y	MEAN	STDV	NPIXELS
```

So in my case the ```index = -1```, since there are no ":" in the line.

I could add a line to raise an error in case ```index == 0```, but theoretically the index should be greater than 10, right? Because there should be "DatHeader=" first. I think maybe the function should raise an error when ```index <= 10```.